tests/bootstrap.php

```diff
@@ -83,6 +85,58 @@ function wc_test_includes() {
 	require_once $wc_tests_framework_base_dir . '/framework/helpers/class-wc-helper-settings.php';
 }
 
+/**
+ * Re-initialize the dependency injection engine.
+ *
+ * The dependency injection engine has been already initialized as part of the Woo initialization, but we need
+ * to replace the registered read-only container with a fully configurable one for testing.
+ * To this end we hack a bit and use reflection to grab the underlying container that the read-only one stores
+ * in a private property.
+ *
+ * Additionally, we replace the legacy/function proxies with mockable versions to easily replace anything
+ * in tests as appropriate.
+ *
+ * @throws \Exception The Container class doesn't have a 'container' property.
+ */
+function wc_blocks_initialize_dependency_injection() {
```

Comment:
This was copied from the WooCommerce core tests bootstrap file. It registers `MockableLegacyProxy::class` in place of `LegacyProxy::class` for the purpose of tests and fixes the errors we were getting with running our phpunit tests.

Reply:
I don't full understand what is happening here, but assuming tests pass 👍🏻